cumulus/polkadot-parachain/polkadot-parachain-lib/src/command.rs

```diff
@@ -215,7 +212,24 @@ pub fn run<CliConfig: crate::cli::CliConfig>(cmd_config: RunConfig) -> Result<()
 				RelayChainCli::<CliConfig>::new(runner.config(), cli.relay_chain_args.iter());
 			let collator_options = cli.run.collator_options();
 
+			let is_dev_chain = runner.config().chain_spec.id().ends_with("-dev");
```

Comment:
I am a bit unsure as to why this limitation should exist. So this will only work if the chain-spec's id ends with -dev. Why should this limitation exist?

Reply:
I will investigate and get back with the details, but the general problem is that I'm getting errors at block production when trying to use a non-dev runtime (for example: `asset-hub-polkadot`).